hoomd/snapshot.py

```diff
@@ -109,3 +109,9 @@ def replicate(self, nx, ny, nz):
 
     def _broadcast_box(self):
         self._cpp_obj._broadcast_box(self._comm.cpp_mpi_conf)
+
+    @classmethod
+    def _from_gsd_snapshot(cls, gsd_snapshot, communicator):
+        sp = cls()
+        snapshot = sp._from_cpp_snapshot(gsd_snapshot, communicator)
```

Comment:
While this works, I feel this may be a bit to fragile. Doesn't really create a standard snapshot, but merely wraps the gsd snapshot with a hoomd snapshot. We want to transfer the data to a C++ snapshot too. Something like
```python
_from_gsd_snapshot(cls, gsd_snapshot, communicator):
    snap = cls()
    snap.particles.N = gsd_snapshot.particles.N
    snap.particles.position[:] = snap.particles.position[:]
```
is more what I was thinking. It is more verbose and work, but I think this will be less likely to break expectations. For instance, I think this would fail if passed in to `create_state_from_snapshot` since the `self._cpp_obj` is not a `_hoomd.Snapshot` object.

Reply:
Yeah, your idea is much better and more understandable! I'm thinking all particle, bond, and box attributes should be set. I'm not sure what to do with the communicator.